### PR TITLE
Hotfix: core/HLE: Fix cellMsgDialog null pointer dereference

### PR DESCRIPTION
**Pull request by TGEnigma**
msgString would be dereferenced even when it is null because the rsx overlay code was put in before the argument checks.
Fixes Sonic Adventure regression.

### CODE REVIEW

**Comment by Megamouse** [comment]
WIP

**Comment by TGEnigma** [comment]
should be good now

File: rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp [hunk 1]
```diff
@@ -320,7 +320,14 @@ s32 cellMsgDialogProgressBarSetMsg(u32 progressBarIndex, vm::cptr<char> msgStrin
 		if (auto dlg2 = rsxthr->shell_get_current_dialog())
 		{
 			if (auto casted = dynamic_cast<rsx::overlays::message_dialog*>(dlg2))
+			{
+				if (!msgString)
```
Comment (kd-11) [inline rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: There is a parameter somewhere after this block, move it to before the GSRender check.

Comment (TGEnigma) [reply rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: I had been told that MsgDialogBase doesn't exist when the overlay is active

Comment (kd-11) [reply rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: Tyhe block from L341-343 has nothing to do with MsgDialogBase. Just move the check up before attempting to create the native interface.

Comment (TGEnigma) [reply rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: I figured that keeping the order of the errors was important assuming someone went out of their way to check the precedence, is it not?

Comment (kd-11) [reply rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: No. The other error (not_opened) means the system could not create the dialog and is impossible on rpcs3 currently.

**Pull request merged**